src/app/InteractionModelEngine.cpp

```diff
@@ -1793,6 +1793,13 @@ CommandHandlerInterface * InteractionModelEngine::FindCommandHandler(EndpointId
     return nullptr;
 }
 
+InteractionModel::DataModel * InteractionModelEngine::SetDataModel(InteractionModel::DataModel * model)
```

Comment:
This namespacing is still really weird.  DataModel does not really live under InteractionModel....

Reply:
It is bad (and forced separate PRs due to clash with protocols::InteractionModel).

Not sure what a better place is though and replacing should probably be a separate change for rename only. If you have a better name idea I can make a separate PR. Otherwise I am leaning towards defering that while the implementations are on-going.

Reply:
Well, we could have `chip::app::DataModel`, but that's an existing namespace.

We could call the class `chip::app::DataModel::Instance`?

Reply:
`Instance` is not a good class name because we have a lot of singleton Instance methods. `Provider` ?

Reply:
Provider seems fine.

Separate PR for this part, obviously.

Reply:
Will have a followup on this.